### PR TITLE
[WIP] Xamarin Forms for WPF

### PR DESCRIPTION
**Pull request by nickrandolph**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Feature

### :arrow_heading_down: What is the current behavior?
N/A

### :new: What is the new behavior (if this is a feature change)?
Adding WPF support for Xamarin Forms

### :thinking: Checklist before submitting

- [ ] All projects build
- [ ] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [ ] Rebased onto current develop

### CODE REVIEW

File: Directory.build.props [hunk 1]
```diff
@@ -24,14 +24,19 @@
     <DebugType>full</DebugType>
     <IsTestProject>$(MSBuildProjectName.Contains('UnitTest'))</IsTestProject>
     <IsLibraryProject>$(MSBuildProjectName.Contains('MvvmCross'))</IsLibraryProject>
+    <IsForms>$(MSBuildProjectName.Contains('.Forms'))</IsForms>
   </PropertyGroup>
   
   <ItemGroup Condition="'$(IsLibraryProject)' == 'true'">
-    <PackageReference Include="MSBuild.Sdk.Extras" Version="1.2.2" PrivateAssets="All" />
+    <PackageReference Include="MSBuild.Sdk.Extras" Version="1.3.0" PrivateAssets="All" />
```
Comment (martijn00) [inline Directory.build.props]: Please do this update in a separate PR.

File: MvvmCross.Forms/MvvmCross.Forms.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">netstandard2.0;Xamarin.iOS10;Xamarin.Mac20;MonoAndroid80;uap10.0.16299</TargetFrameworks>
-    <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">netstandard2.0;Xamarin.iOS10;Xamarin.Mac20;MonoAndroid80</TargetFrameworks>
+    <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">netstandard2.0;net461;uap10.0.16299</TargetFrameworks>
```
Comment (martijn00) [inline MvvmCross.Forms/MvvmCross.Forms.csproj]: On Windows we need to build everything, because that is where the CI runs. On Mac we cannot build UWP, WPF and Tizen.

File: MvvmCross.Forms/MvvmCross.Forms.csproj [hunk 3]
```diff
@@ -24,6 +24,9 @@
     <Compile Include="Platforms\Net\**\*.cs" />
     <Compile Include="Platforms\Wpf\**\*.cs" />
     <Compile Include="Platforms\Console\**\*.cs" />
+    <PackageReference Include="System.Console" Version="4.3.0" />
```
Comment (martijn00) [inline MvvmCross.Forms/MvvmCross.Forms.csproj]: Isn't this already in the main library?

File: Directory.build.props (hunk 1 above)
Comment (nickrandolph) [reply Directory.build.props]: yep, no probs - it'll need to happen before v3 of XF that's all. I suspect this whole PR will be broken into chunks and then this one abandoned

File: MvvmCross.Forms/MvvmCross.Forms.csproj (hunk 2 above)
Comment (nickrandolph) [reply MvvmCross.Forms/MvvmCross.Forms.csproj]: Yeh, sorry this one slipped past me and shouldn't be been committed

File: MvvmCross.Forms/MvvmCross.Forms.csproj (hunk 3 above)
Comment (nickrandolph) [reply MvvmCross.Forms/MvvmCross.Forms.csproj]: yeh, I was having some issues just getting it all to build. It should be able to be removed now

**Comment by flyingxu** [comment]
I am using WPF so I hope this can be merged now, but I understand the concern of using a `pre` version of Xamarin.Form in current MvvmCross.Forms.

We are using 'multi-target' build configuration, is there any possibility that we configure the `.csproj`s so that the `pre` version of Xamarin.Forms is only used for `MvvmCross.Forms` `net461` build? By doing this, other build targets will not be impacted so this PR can be merged, and users want to test Xamarin.Wpf with MvvmCross.Forms can test now.

**Comment by nickrandolph** [comment]
Unfortunately I don’t think that’s going to be possible. The multi-targeting tooling is so fragile that I think this will result in some very weird behaviour and too risky to do at this stage. Not to mention that the v3 XF targets seems to have a build issue at the moment.
This PR is here just so that we can track progress and see what changes will be required as v3 matures. Aim is to be ready to go soon after it drops.
This will also not get back ported to <v6 of mvvmcross

**Comment by martijn00** [comment]
The code in develop has changed so much that this isn't very compatible anymore. I'll close it for now, but when Forms 3.0 hits we can open a new PR based on this code.
